src/lib/pipeline.ml

```diff
@@ -165,13 +165,33 @@ module Full (Bfx: Extended_edsl.Semantics) = struct
         in
         sname, Stdlib.fastq_of_input i)
 
+  (** 
+     A record to join a value and its saved version.
+
+     Previous versions of Eidisco had a [Bfx.save] construct of type
+     ['a repr -> 'a repr] which would “hide” saved values, we would
+     then put [Bfx.save ~name] calls everywhere and forget about
+     them. The new [save] from Biokepi is explicit, so to avoid too
+     much rewriting of this pipeline we pack values together.
+
+  *)
+  type 'a saved = {
+    content: 'a Bfx.repr;
+    saved: [ `Saved of 'a ] Bfx.repr;
+  }
+
+  (** This is like {!Biokepi.EDSL.Semantics.save} but packs the saved
+      node with its original value.  *)
+  let save_through ~name x =
+    { content = x; saved = Bfx.save ~name x }
 
   let seq2hla_hla fqs =
-    Bfx.seq2hla (Bfx.concat fqs) |> Bfx.save ~name:"Seq2HLA"
+    Bfx.seq2hla (Bfx.concat fqs) |> save_through ~name:"Seq2HLA"
 
 
   let optitype_hla fqs ftype name =
-    Bfx.optitype ftype (Bfx.concat fqs) |> Bfx.save ~name:("OptiType-" ^ name)
+    Bfx.optitype ftype (Bfx.concat fqs)
+    |> save_through ~name:("OptiType-" ^ name)
```

Comment:
Read your comment above, but I still think `Bfx.save` was more verbose/readable compared to the `save_through` solution. I like all the new fancy stuff that comes with the new provenance solution, but wish we had a similar way of expressing this save operation :man_shrugging: